### PR TITLE
Add Connections, Secrets and Variables to Airflow Helm Chart documentation

### PR DESCRIPTION
**Pull request by drdenzy**
This PR aims to further improve Airflow Helm chart documentation.

It illustrates how connections, secrets, and environment variables can
be passed into Airflow using Helm chart.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+Adding Connections, Secrets and Environment Variables
+==================================================
+
+You can programmatically add Connections, Secrets and Environment Variables to your
+airflow deployment using Helm chart.
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
You can programmatically add Connections, Variables and arbitrary Environment Variables to your
Airflow deployment using the Helm chart.
```

I'd probably leave the k8s secret piece out of the mix here, personally.

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+         AIRFLOW_CONN_AWS: 'base64_encoded_aws_conn_string'
+       stringData: |
+         AIRFLOW_CONN_OTHER: 'other_conn'
+     {{ .Release.Name }}-other-secret-name-suffix: |
+       data: |
+
+
+Variables
+---------
+Airflow supports Variables which enable users craft dynamic DAGs. You can set variables in Airflow in three ways - UI,
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
Airflow supports Variables which enable users to craft dynamic DAGs. You can set Variables in Airflow in three ways - UI,
```

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+commandline, and within your DAG file. See:doc:`/howto/variable`.` for more.
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
commandline, and within your DAG file. See :doc:`/howto/variable` for more.
```
I think this is the right format for the link?

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+Adding Connections, Secrets and Environment Variables
+==================================================
```
Comment (kaxil) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
Adding Connections, Secrets and Environment Variables
=====================================================
```

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+     {{ .Release.Name }}-other-secret-name-suffix: |
+       data: |
+
+
+Variables
+---------
+Airflow supports Variables which enable users craft dynamic DAGs. You can set variables in Airflow in three ways - UI,
+commandline, and within your DAG file. See:doc:`/howto/variable`.` for more.
+
+With Helm chart, you can also inject environment variables into airflow. For example, in a yaml file to
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
With the Helm chart, you can also inject environment variables into Airflow. For example, in a yaml file to
```

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+override values of interest in the ``env`` section of the ``values.yaml`` file.
+
+.. code-block:: yaml
+
+   env:
+     - name: "AIRFLOW_VAR_KEY"
+       value: "value_1"
+
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
```

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+You can also utilize the ``extraEnv`` and ``extraEnvFrom`` to set more possibly "templatable" values for your airflow
+deployment.
+
+.. code-block:: yaml
+
+   extraEnv: |
+     - name: AIRFLOW__CORE__LOAD_EXAMPLES
+       value: 'True'
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
     - name: AIRFLOW_VAR_HELM_RELEASE_NAME
       value: '{{ .Release.Name }}'
```

Might as well have an example that uses templating.

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+   env:
+     - name: "AIRFLOW_VAR_KEY"
+       value: "value_1"
+
+     - name: "AIRFLOW_VAR_ANOTHER_KEY"
+       value: "value_2"
+
+
+You can also utilize the ``extraEnv`` and ``extraEnvFrom`` to set more possibly "templatable" values for your airflow
+deployment.
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
You can also utilize ``extraEnv`` and ``extraEnvFrom`` if you need the name or value to be templated.
```

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+   extraEnv: |
+     - name: AIRFLOW__CORE__LOAD_EXAMPLES
+       value: 'True'
+
+
+   extraEnvFrom: |
+     - secretRef:
+         name: '{{ .Release.Name }}-airflow-connections'
+     - configMapRef:
+         name: '{{ .Release.Name }}-airflow-variables'
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
   extraEnvFrom: |
     - configMapRef:
         name: '{{ .Release.Name }}-airflow-variables'
         
   extraConfigMaps:
     '{{ .Release.Name }}-airflow-variables':
       data: |
         AIRFLOW_VAR_HELLO_MESSAGE: "Hi!"
```
Might as well be a complete working example.

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+
+Adding Connections, Secrets and Environment Variables
+==================================================
+
+You can programmatically add Connections, Secrets and Environment Variables to your
+airflow deployment using Helm chart.
+
+
+Connections and Secrets
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
Connections and Sensitive Environment Variables
```

Maybe?

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+-----------------------
+Under the ``secret`` and ``extraSecret`` sections of the ``values.yaml`` you can pass connection strings and secrets
+into Airflow using Helm chart. To illustrate, lets create a yaml file to override values under these sections of the
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
Under the ``secret`` and ``extraSecret`` sections of the ``values.yaml`` you can pass connection strings and sensitive environemt variables
into Airflow using the Helm chart. To illustrate, lets create a yaml file to override values under these sections of the
```

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+  extraSecrets:
+     {{ .Release.Name }}-airflow-connections:
+       data: |
+         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
+         AIRFLOW_CONN_AWS: 'base64_encoded_aws_conn_string'
+       stringData: |
+         AIRFLOW_CONN_OTHER: 'other_conn'
+     {{ .Release.Name }}-other-secret-name-suffix: |
+       data: |
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
   secret:
     - envName: "AIRFLOW_CONN_GCP"
        secretName: "my-airflow-connections"
        secretKey: "AIRFLOW_CONN_GCP"
      - envName: "my-env"
        secretName: "my-secret-name"
        secretKey: "my-secret-key"

  extraSecrets:
     my-airflow-connections:
       data: |
         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
     my-secret-name: |
       stringData: |
         my-secret-key: my-secret
```

Have it be a working example, and show both an env var and a connection.

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+Adding Connections, Secrets and Environment Variables
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
Adding Connections, Variables and Environment Variables
```

I wouldn't consider `Secrets` to be core concept 🤷‍♂️

**Review by jedcunningham — CHANGES_REQUESTED** [review]

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+You can also utilize the ``extraEnv`` and ``extraEnvFrom`` to set more possibly "templatable" values for your airflow
+deployment.
+
+.. code-block:: yaml
+
+   extraEnv: |
+     - name: AIRFLOW_VAR_HELM_RELEASE_NAME
+       value: '{{ .Release.Name }}'
+
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst:77]:
```suggestion
```
Less vertical space.

**Review by jedcunningham — CHANGES_REQUESTED** [review]

File: docs/helm-chart/adding-connections-and-variables.rst (hunk 8 above)
Comment (jedcunningham) [reply docs/helm-chart/adding-connections-and-variables.rst]: Bump, was this accidentally missed?

**Review by jedcunningham — APPROVED** [review]
LGTM!

**Review by kaxil — APPROVED** [review]

File: docs/helm-chart/adding-connections-and-variables.rst [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+         AIRFLOW_CONN_GCP: 'base64_encoded_gcp_conn_string'
+     my-secret-name: |
+       stringData: |
+         my-secret-key: my-secret
+
+
+Variables
+---------
+Airflow supports Variables which enable users to craft dynamic DAGs. You can set Variables in Airflow in three ways - UI,
+commandline, and within your DAG file. See :doc:`/howto/variable` for more.
```
Comment (jedcunningham) [inline docs/helm-chart/adding-connections-and-variables.rst]:
```suggestion
cli, and within your DAG file. See :doc:`apache-airflow:howto/variable` for more.
```

Should fix the spelling and link issue.

**Pull request merged**